src/components/TableFilter.js

```diff
@@ -310,13 +308,20 @@ class TableFilter extends React.Component {
           </div>
           <div className={classes.filtersSelected} />
         </div>
-        {options.filterType === 'checkbox'
-          ? this.renderCheckbox(columns)
-          : options.filterType === 'multiselect'
-          ? this.renderMultiselect(columns)
-          : options.filterType === 'textField'
-          ? this.renderTextField(columns)
-          : this.renderSelect(columns)}
+        <GridList cellHeight="auto" cols={2}>
+          {columns.map((column, index) => {
+            if (column.filter) {
```

Comment:
Since you copied my `if` here you should remove all ternaries doing the same verification in each sub renders 😉

Reply:
I'm sorry, I didn't get your point here.
BTW, I saw your PR only after I started implementing it. So raised a PR anyway. Hope it's fine😃

Reply:
at this line if the filter boolean in true or false to show the filter and you check it again at lines 197 , 229 and 251.
Aside of that I think your integration with the `Grid` is much better than what I did 👏 
Also I copied your testing part to make my built pass and I want to thank you for that 😄

Reply:
Fixed it. Thanks for pointing it out.